### PR TITLE
7 UI structure

### PR DESCRIPTION
**Pull request by nestorbe**
Features implemented so far (Rinkeby Test Network):

**StableRedemption.sol**
- Ability to redeem dOrg token 1:1 for a stablecoin.
- dOrg token balance check.

https://user-images.githubusercontent.com/51724662/103048847-f0410480-455d-11eb-93cc-bfe10b2a78eb.mp4

Pending features:

**StableRedemption.sol**
- Ability to redeem dOrg token 1:1 for multiple stablecoins.

**StakingReward.sol**
- dOrg token stake ability.
- dOrg token unstake ability.
- dOrg token claim rewards ability.
- dOrg token unstake and claim ability.

### CODE REVIEW

**Comment by dOrgJelli** [comment]
Looks great! Here are some points of feedback. Since this app is meant for internal dOrg use only, I wouldn't worry too much about these, but just thoughts on best practices:
- Addresses of deployed contracts are hard-coded. These should be pulled from some configuration file that keeps track of contract deployments. This way when you run `deploy:mainnet`, the config file is updated with new address, and the app automatically works.
- Move the contracts into their own package, maybe in a folder named "protocol". Then you can make the root folder a monorepo where the client imports the protocol as a dependency. The addresses.json file described above can be published as an artifact in the protocol package. Names of the packages could be "@dorgtech/dorg-token-contracts" and "@dorgtech/dorg-token-ui".
- The Web3 provider instance is an `any` type, which can be dangerous. Checkout this class I created that wraps Ethers and is fully typed. It supports Web3 provider JS objects (metamask, etc), RPC URL strings, and in-memory private keys: https://github.com/Web3-API/prototype/blob/579d2dc860d1eb269da09487de68ef8267ef2144/packages/client-js/src/portals/Ethereum.ts#L1
- In the UI's dependencies, there are a few things that are actually devDependencies (`@types/...` for example).

File: client/.eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...dOrg/token-redemption/client/src/containers/Redemption.tsx",["37"],"/home/horsecat/dOrg/token-redemption/client/src/containers/index.ts",[],"/home/horsecat/dOrg/token-redemption/client/src/containers/Header.tsx",["38"],{"ruleId":"39","replacedBy":"40"},{"ruleId":"41","replacedBy":"42"},{"ruleId":"43","severity":1,"message":"44","line":79,"column":3,"nodeType":"45","endLine":79,"endColumn":12,"suggestions":"46"},{"ruleId":"47","severity":1,"message":"48","line":2,"column":10,"nodeType":"45","messageId":"49","endLine":2,"endColumn":21},"no-native-reassign",["50"],"no-negated-in-lhs",["51"],"react-hooks/exhaustive-deps","React Hook useEffect contains a call to 'setConfMessage'. Without a list of dependencies, this can lead to an infinite chain of updates. To fix this, pass [inputTokenAmount, stablecoin, stablename] as a second argument to the useEffect Hook.","Identifier",["52"],"@typescript-eslint/no-unused-vars","'getProvider' is defined but never used.","unusedVar","no-global-assign","no-unsafe-negation",{"desc":"53","fix":"54"},"Add dependencies array: [inputTokenAmount, stablecoin, stablename]",{"range":"55","text":"56"},[2297,2297],", [inputTokenAmount, stablecoin, stablename]"]
```
Comment (dOrgJelli) [inline client/.eslintcache:1]: Should be git ignored

File: scripts/deploy.ts [hunk 2]
```diff
@@ -23,15 +23,13 @@ async function main() {
...
-      ...config
-    );
+    await sc.initialize(config.owner, config.inputToken, config.redemptionTokens)
 
     log[contract].initializeParams = {
       ...config
     };
 
-    logDeployment(log, contract);
+    //logDeployment(log, contract);
```
Comment (dOrgJelli) [inline scripts/deploy.ts:32]: Can this be uncommented, or is it breaking somewhere?

File: scripts/common.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export async function deployAndVerify(
   await new Promise((resolve) =>
     setTimeout(() => resolve(), 30000)
   );
+  /**
```
Comment (dOrgJelli) [inline scripts/common.ts:18]: If this is breaking w/o the etherscan API key, can we just make it optional and only run the `exec` if the ENV var is present?

**Review by dOrgJelli — CHANGES_REQUESTED** [review]

File: scripts/common.ts (hunk 3 above)
Comment (nestorbe) [reply scripts/common.ts:18]: This is breaking with the API key, this is the reason it's giving me: **Failed to verify contract. Reason: Max rate limit reached, please use API Key for higher rate limit**

File: scripts/deploy.ts (hunk 2 above)
Comment (nestorbe) [reply scripts/deploy.ts:32]: fixed this, I commented it for testing...

File: client/.eslintcache (hunk 1 above)
Comment (nestorbe) [reply client/.eslintcache:1]: done!

File: scripts/common.ts (hunk 3 above)
Comment (dOrgJelli) [reply scripts/common.ts:18]: Ah got it. Okay let's just leave this commented for now :P

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**